### DISCUSSION TITLE
Qual a melhor distro para Linux?

### DISCUSSION THREAD

**Discussion by jaocarli** [frontendbr/forum#1422] [Discussão]
Olá meus queridos(as)! Como estão? Então sou iniciante no Frontend, estou tendo dificuldades com algumas coisas no Windows, e quebrando a cabeça para tentar instalar o Laravel, que por sinal, somente da erro e dor de cabeça... e olha que é apenas o meu primeiro job... como não tenho grana para bancar um MacBook, pensei em instalar alguma distro Linux no meu computador para usar apenas para trabalho, deixando assim o Windows para programas de Design (na qual é a minha formação). Para vocês, qual a melhor distro para o Front? 

Abraços! :D 
Edit 1: Não sei como colocar tags. Mil perdões!

**Comment by onlurking** [comment]
Booaa! Então, eu diria que depende do teu objetivo, a aposta mais tranquila que tu pode fazer é pegar uma distro que usa o **apt** como gerenciador de pacotes(é como é chamado o software que instala e atualiza os programas), nessa categoria está o Debian, o Ubuntu e o Linux Mint(recomendo o Mint acima do Ubuntu), isso porque ambos tem vários tutoriais em pt-br e se você tiver algum problema vai ficar bem confortável pra resolver.

Agora se tu tá com o objetivo de pegar uma distro e aprender Linux pra valer eu recomendo ir de [Manjaro XFCE](https://manjaro.org/download/xfce), ele é baseado no Arch Linux que é uma distro considerada avançada, o motivo de eu estar recomendando o Manjaro é por causa da [Arch Wiki](https://wiki.archlinux.org), que apesar de ser do Arch, é 99% aplicável ao Manjaro e ela tem um nível técnico muito alto, se você tiver algum problema você provavelmente vai cair nela e vai aprender MUITO lendo e resolvendo. E os programas! Manjaro tem muitos programas no repositório, e se não estiver então vai estar nos repos não oficiais(AUR), o Ubuntu também tem um repo não oficial (que é Ubuntu PPA) pros softwares que não estão presentes, mas é muito mais pobre que a AUR do Manjaro/Arch Linux.

Uma distro Linux é basicamente composta por:
- o gerenciador de pacotes, que é como você instala, remove e atualiza os programas nelas
    - ubuntu, debian e mint usam **apt**, arch e manjaro usam o **pacman**
- o repositório, que é um servidor que o gerenciador de pacotes pega os programas e as bibliotecas(chamado de pacotes)
    - cada distro costuma manter seus próprios servidores e com isso seus pacotes(e as versões diferem), sendo assim o ubuntu por exemplo costuma ter mais pacotes atualizados que o debian, e o manjaro e principalmente o arch linux possuem pacotes extremamente atualizados(mais do que todas as outras distros, isso tem prós e contras).
- os programas que vem por padrão, e pasme, a interface é um programa que pode ser trocado numa distro, inclusive você pode ter várias interfaces
    - tem algumas um pouco mais pesadas como KDE, Gnome, Cinnamon, são bonitas, entretanto não são tão rápidas comparadas com as interfaces mais leves
    - tem umas extremamente leves, como o XFCE(eu uso essa há anos), então minha escolha é sempre Manjaro XFCE e Mint XFCE(que usam essa interface por padrão)
- se é rolling release ou point release, point release quer dizer que ou os desenvolvedores da distro seguram as atualizações e de tempos em tempos eles lançam uma versão atualizada do OS, Ubuntu é assim, então ele tem as versão 18.04, 18.10, 19.04 e tem as rolling release aonde é lançado atualizações constantemente, e se você atualiza seu sistema você sempre fica sempre última versão(eu particularmente prefiro rolling release visto que não tenho que reinstalar a distro de meses em meses ou fazer uma migração manual de versão que pode ser perigosa e quebrar tudo, já aconteceu algumas vezes comigo).

Tem mais outros detalhes, mas a sacada é você explorar e ver as distros que lhe agradam nessas partes, tudo é gosto, eu particularmente prefiro tudo atualizado e desafio técnico(entretanto não tão técnico a ponto de atrapalhar minha produtividade como desenvolvedor), então Manjaro caí como uma luva.

**Comment by valkervieira** [comment]
Ótima decisão! Tu não vai se arrepender.

Tem um clichê que é super válido quando o assunto é escolher uma distribuição: Não existe melhor distro, existe a melhor distro pra você! Por mais que tu leia recomendações, a distro ideal pra você vai ser aquela que tu vai testar e se sentir em casa usando, abraça a liberdade que o Linux te dá e testa várias!

Eu por muito tempo me forcei a usar linux baseadas em Ubuntu, que é uma distribuição super popular, usei **Elementary OS**, Mint e o **Ubuntu** propriamente dito, eu não me sentia confortável e parecia que eu tava me forçando a usar. Mas pode ser que com você seja diferente! :smile: 

Eu me encontrei em distros baseadas em **Arch** linux! O Arch é uma distribuição um pouco diferente, por que você precisa instalar tudo manualmente, desde o particionamento, até montar, preparar o computador para bootar, instalar o desktop, os drivers e assim vai... 

Depois de um tempo eu fiquei de saco cheio de fazer isso e encontrei o **Antergos**, que é como se fosse um "instalador visual do Arch", você instala o sistema, escolhe qual desktop você quer, e tem a experiência Arch crua, num sistema com praticamente nada instalado pra você personalizar do seu jeito, usei o Antergos por bastante tempo, enjoei, agora estou há uns 2 meses no **Manjaro** e estou amando! Provavelmente vou ficar nela por bastante tempo também (ou até enjoar rs).

Algo que é importante ponderar é quão poderosa é tua máquina, principalmente na hora de escolher qual desktop tu vai usar, DEs como o **Gnome** e o **KDE** são um pouco pesadas e exigem um certo poder gráfico e de processamento para funcionarem legal, outras como o **XFCE** ou **LXDE** são mais leves porém com menos possibilidade de customização.

Eu valorizo bastante o visual do sistema e minha máquina aguenta qualquer desktop, então eu usei por bastante tempo o Gnome que é bem pesado personalizado pra ficar com uma carinha do Mac, agora uso o KDE que é um pouco menos pesado e permite tanta personalização quanto o Gnome. Nesse momento eu estou com umas 10 abas abertas no Chromium, um app em React rodando com live reload e o VS Code aberto, não passa de 3GB de uso de Ram, o que pra mim ta ótimo!

Minha recomendação é: teste o Manjaro KDE, mas se não gostar não se preocupa, só testar outro até achar a que tu vai gostar mais!

**Comment by jaocarli** [comment]
Opa! Muuuuuito obrigado pelas respostas @borghettinho e @onlurking! De início vou testar o **Elementary OS** para ver como vai ser e se vou me acostumar. Caso isso não aconteça, vou migrar para o Manjaro, como ambos recomendaram! 
Estou pensando em tirar o dia para fazer essa migração, tendo em vista que vou formatar meu windows e repartir o SSD, para assim instalar as distros e os programas que irei precisar. Vou usar o Windows somente para usar os meus programas de Design e alguns jogos

**Comment by jaocarli** [comment]
Mas infelizmente, não deu. Tentei o Elementary OS 5, Ubuntu 18 e o Ubuntu 19, mas todos travam na mesma tela, e o ubuntu 19 sequer reconhece o meu mousepad e mouse. Vou ficar pelo windows mesmo :/

**Comment by onlurking** [comment]
@fl4m3x já tentou Manjaro XFCE?

**Comment by jaocarli** [comment]
@onlurking ainda não... vou colocar esse para baixar agora. Aproveitando o comentário, é uma boa deixar, mais ou menos, quantos gb de espaço para instalar distros linux? tendo em vista que vou apenas trabalhar na mesma.

**Comment by onlurking** [comment]
Cara, depende, eu sempre deixo o HD todo já que só uso Linux, mas pra dual boot e se for pra trabalho, eu deixaria pelo menos uma folga boa, tipo uns 200GB se tu tiver um HD de 500GB, geralmente eu faço isso:

Uma partição **FAT32** com 512MB montada como `/boot/efi` (e com as flags `boot` e `esp`)
Uma partição **linuxswap** com o tamanho da sua RAM (eg, 8096MB se tu tiver 8GB de RAM)
Uma partição **EXT4** montada como `/` com um tamanho decente(O linux todo vai ser instalado aqui)

**Comment by valkervieira** [comment]
Então, pelo jeito você está tentando instalar junto com o windows em dual boot certo? Nesse caso eu super recomendo o Manjaro Illirya, ele tem uma opção de instalaçao ao lado de outro sistema e que pra mim sempre funcionou bem.

Quanto ao espaço eu acho que depende do teu uso e do quanto de espaço tu tem no HD, em casa eu também só uso Linux então não me preocupo e instalo em todo SSD, mas aqui no trabalho eu tenho um HD de 500GB, 70GB para o `/` 512MB para o `/boot/efi` e 8GB de `swap`, exatamente como o @onlurking, o resto para o Windows.

Não desiste não, pergunta ai pra nós, no começo sempre alguma coisa da um problema. Eu mesmo até hoje sempre que instalo preciso fazer uma gambiarra pro wi-fi do meu note funcionar. :sweat_smile:

**Comment by wkrueger** [comment]
Eu ando pendendo para distribuições ubuntu-based basiconas (ubuntu/mint), pelo simples fato de haver um suporte mais sólido da comunidade em geral. No linux você já tenderá a gastar tempo improdutivo  com instalações não suportadas de coisas, e usando um ubuntu-based você deve reduzir isso.

Só fuja da DE padrão do ubuntu (Unity) que é horrível (isto você pode trocar a qualquer momento sem muito trabalho).

Em segundo lugar eu gostei do Arch/Manjaro, mas o seu sistema de rolling release pode trazer ocasionais dores de cabeça ao atualizar pacotes.

**Comment by valkervieira** [comment]
@wkrueger o Unity não é mais o desktop padrão do Ubuntu (graças a deus) desde o Bionic Beaver / 18 LTS, desde então é Gnome, e ta muito bom.

Essa questão da comunidade é importante, mas tem bastante material do Arch também, sem contar que a Wiki é muito completa ao ponto de normalmente as questões não precisam ser perguntadas/respondidas por que já estão muito bem documentadas por lá.

E por último a questão do rolling release dar pau é justamente um dos principais motivos do Manjaro ser tão recomendado, ele te oferece o melhor do Arch (AUR, pacman, controle) atrasando um pouquinho os updates que poderiam quebrar o sistema, alguns pacotes inclusive avisam que updates no Manjaro são atrasados. Eu mesmo uma vez recebi um update no Arch puro que meu wi fi simplesmente parou de funcionar (eu precisava de modprobe), instalei o Manjaro, atualizei tudo e esse update simplesmente não tava lá, menos de uma semana depois o update chegou já com um hotfix e o meu wi fi não deu problema nenhuma vez.

**Comment by wkrueger** [comment]
Eu tinha um manjaro que eu logava muito pouco (dual boot) e frequentemente dava conflitos em updates que tinha que cavar no google pra resolver, e em um último momento um update me deixou em grub rescue. Obviamente fiquei em fúria e troquei de distro.

**Comment by evertonthepaula** [comment]
A melhor distro Linux é aquela que você domina, então já que você está vindo do windows, para começar escolha uma que vai te dar uma curva de aprendizado pequena, eu comendaria Ubunutu. Por quê? Bem você vai gastar muitas horas da sua vida masterizando linux e o resto dela masterizando front-end, você não pode ser dar ao luxo de gastar muito tempo agora com uma distro complicada, você não deve procurar a "melhor distro", você deve procurar por uma distro que vai te ensinar de uma forma muito simples como o linux funciona, depois aí sim você procura distro mais poderosa de todas que vai consumir sua vida para aprender a funcionar, mas quando vc terminar vai ser lindo.

**Comment by nandomoreirame** [comment]
não existe melhor nem pior, existe disto que vc curte e que resolve seus problemas.
é muito mais uma relação de amor e ódio do que outra coisa.

**Comment by adeonir** [comment]
é igual IDE, não existe melhor ou pior

**Comment by nandomoreirame** [comment]
O ser humano gosta muito de criar polêmicas e polarizar qualquer coisa, é bem comum... tudo precisa ter o lado ruim e o lado bom... o que eu estou usando sempre é o melhor pq fui eu que escolhi.

**Comment by jaocarli** [comment]
Após muita briga e quase jogar meu notebook janela (e sete andares abaixo), consegui instalar o ElementaryOS, após instalar o Manjaro e não ter me acostumado muito :/ agora é ver o que vai rolar auehauhe obrigado todos os comentários!

**Comment by moacirjun** [comment]
Boa, galera!! Vocês são demais.
Acabei de instalar aqui o Linux Mint 19.1 "Tessa" - Xfce graças ao que li aqui.
Estou vindo do Ubuntu 18.04.2 e até então estou curtindo e (o mais importante) sobrevivendo kkk.
Depois de usar mais o Mint pretendo trazer aqui minha opinião sobre o que achei.
Valeuuuu!!
